src/xtgeo/common/sys.py

```diff
@@ -260,118 +306,107 @@ def resolve_alias(self, obj):
 
         self._file = (parent / newname).with_suffix(suffix)
 
-    def exists(self):  # was: file_exists
-        """Check if 'r' file, memory stream or folder exists, and returns True of OK."""
+    def exists(self) -> bool:
+        """Returns True if 'r' file, memory stream, or folder exists."""
         if "r" in self._mode:
-            if isinstance(self._file, io.BytesIO):
-                return True
-
-            if self._file.exists():
+            if isinstance(self.file, (io.BytesIO, io.StringIO)):
                 return True
+            return self.file.exists()
 
-            return False
-
+        # Writes and appends will always exist after writing
         return True
 
-    def check_file(self, raiseerror=None, raisetext=None):
-        """Check if a file exists, and raises an OSError if not.
+    def check_file(
+        self,
+        raiseerror: Optional[Type[Exception]] = None,
+        raisetext: Optional[str] = None,
+    ) -> bool:
+        """
+        Check if a file exists, and raises an OSError if not.
 
         This is only meaningful for 'r' files.
 
-        Args:
-            raiseerror (Exception): Type of Exception, default is None, which means
-                no Exception, just return False or True
-            raisetext (str): Which message to display if raiseerror, None gives a
-                default message.
+        Parameters:
+            raiseerror:
```

Comment:
Is `Parameters:` better than `Args:` for google style docstrings?

Reply:
https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html

Reply:
It's worse in that it's more characters, but `Args` is actually an alias of `Parameters` ([noted in here](https://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html)). I'm just being pedantic because technically parameters is the right term and in the compiled HTML docs pages it will be listed as `Parameters`. numpy docs uses `Parameters` as well.

I don't really have strong feelings about what's better or not, but at least I think we should diverge from Google standards when it comes to giving the type under `Args` in the doc strings. autodoc and autocompletors will just ignore it if the mypy type annotations are there and use those, and since they are actually checked will actually be up to date.

What do you think?